### DISCUSSION TITLE
How to get SMARTS attribute for atoms and bonds after `Chem.MolFromSmarts(smarts)`

### DISCUSSION THREAD

**Discussion by hjuinj** [rdkit/rdkit#7662] [Q&A] [answered]
If I load in a SMARTS as a mol object, is it possible to retrieve the SMARTS attributes (such as these https://www.rdkit.org/docs/RDKit_Book.html#smarts-reference) as properties? E.g. say I have:

```
from rdkit import Chem
mol = Chem.MolFromSmarts("[#6;D1]")
for atm in mol.GetAtoms():
    print(atm.GetSymbol(), atm.GetSmarts())
```
I can only find how to retrieve the explicit degree attribute (`D1`) for the atom as a string output, which I then need to process with some form of custom regex. Is it also possible to get different types of attribute as atom and bond properties, or some other ways of processing these information?

Thank you.

**Comment by greglandrum** [comment, marked as answer]
@hjuinj There's no way to directly do this, but I think you could write some code to parse the results of `DescribeQuery()` from the atoms and bonds and work with that.
Here's what that output looks like:
```
In [6]: mol = Chem.MolFromSmarts("[#6;D3;R2,x3]C")

In [7]: print(mol.GetAtomWithIdx(0).DescribeQuery())
AtomAnd
  AtomAnd
    AtomAtomicNum 6 = val
    AtomExplicitDegree 3 = val
  AtomOr
    AtomInNRings 2 = val
    AtomRingBondCount 3 = val


In [8]: print(mol.GetAtomWithIdx(1).DescribeQuery())
AtomType 6 = val


In [9]: print(mol.GetBondWithIdx(0).DescribeQuery())
SingleOrAromaticBond 1 = val
```

**Reply by hjuinj** [reply]
thanks for the insight Greg!
